src/libstd/thread_local/mod.rs

```diff
@@ -205,6 +205,7 @@ macro_rules! __thread_local_inner {
 
         #[cfg(any(not(any(target_os = "macos", target_os = "linux")), target_arch = "aarch64"))]
         const _INIT: ::std::thread_local::__impl::KeyInner<$t> = {
+            #[allow(unsafe_code)]
             unsafe extern fn __destroy(ptr: *mut u8) {
                 ::std::thread_local::__impl::destroy_value::<$t>(ptr);
```

Comment:
Nice, but do you still need the cast? I'm mainly asking because if you do, that looks like a bug/limitation in the "coerce to expected type" logic in typeck.